src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/UploadContestedCaseDocumentsService.java

```diff
@@ -0,0 +1,221 @@
+            .filter(d -> d.getUploadedCaseDocument().getCaseDocumentParty() != null
+                && d.getUploadedCaseDocument().getCaseDocumentParty().equals(APPLICANT))
+            .filter(d -> isTypeValidForEvidence(d.getUploadedCaseDocument().getCaseDocumentType()))
+            .collect(Collectors.toList());
+        List<ContestedUploadedDocumentData> applicantEvidenceCollection = getDocumentCollection(caseData, APPLICANT_EVIDENCE_COLLECTION);
+        applicantEvidenceCollection.addAll(evidenceInSupportFiltered);
+        log.info("Adding items: {}, to Applicant Evidence In Support Collection", evidenceInSupportFiltered);
+
+        uploadedDocuments.removeAll(evidenceInSupportFiltered);
+
+        caseData.put(APPLICANT_EVIDENCE_COLLECTION, applicantEvidenceCollection);
+    }
+
+    private void filterApplicantTrialBundle(List<ContestedUploadedDocumentData> uploadedDocuments, Map<String, Object> caseData) {
+        List<ContestedUploadedDocumentData> trialBundleFiltered = uploadedDocuments.stream()
+            .filter(d -> d.getUploadedCaseDocument().getCaseDocumentParty() != null
+                && d.getUploadedCaseDocument().getCaseDocumentParty().equals(APPLICANT))
+            .filter(d -> isTypeValidForTrialBundle(d.getUploadedCaseDocument().getCaseDocumentType()))
+            .collect(Collectors.toList());
+        List<ContestedUploadedDocumentData> applicantTrialBundleCollection = getDocumentCollection(caseData, APPLICANT_TRIAL_BUNDLE_COLLECTION);
+        applicantTrialBundleCollection.addAll(trialBundleFiltered);
+        log.info("Adding items: {}, to Applicant Trial Bundle Collection", trialBundleFiltered);
+
+        uploadedDocuments.removeAll(trialBundleFiltered);
+
+        caseData.put(APPLICANT_TRIAL_BUNDLE_COLLECTION, applicantTrialBundleCollection);
+    }
+
+    private void filterRespondentCorrespondence(List<ContestedUploadedDocumentData> uploadedDocuments, Map<String, Object> caseData) {
+        List<ContestedUploadedDocumentData> respondentCorrespondenceFiltered = uploadedDocuments.stream()
+            .filter(d -> d.getUploadedCaseDocument().getCaseDocumentParty() != null
+                && d.getUploadedCaseDocument().getCaseDocumentParty().equals(RESPONDENT))
+            .filter(d -> isTypeValidForCorrespondence(d.getUploadedCaseDocument().getCaseDocumentType()))
+            .collect(Collectors.toList());
+        List<ContestedUploadedDocumentData> respondentCorrespondenceCollection =
+            getDocumentCollection(caseData, RESPONDENT_CORRESPONDENCE_COLLECTION);
+
+        respondentCorrespondenceCollection.addAll(respondentCorrespondenceFiltered);
+        log.info("Adding items: {}, to Applicant Correspondence Collection", respondentCorrespondenceFiltered);
+
+        uploadedDocuments.removeAll(respondentCorrespondenceFiltered);
+
+        caseData.put(RESPONDENT_CORRESPONDENCE_COLLECTION, respondentCorrespondenceCollection);
```

Comment:
whats the use case for storing each of these types in separate collections rather than a single collection?

Reply:
Labels in CCD having "Correspondence", "FR Forms", "Evidence in Support", "Trial Bundle" for both the applicant and respondent. I had assumed when starting this ticket it would be 2 collections, one for app and resp. However Grace explained the use case for this and unfortunately given the tabs these go on, I had to make 8 in total